### ISSUE TITLE
Persist max trace length in `ExecutionTrace`

### ISSUE THREAD

**Issue by frisitano** [0xPolygonMiden/miden-vm#1001]
After a programs execution trace has been generated it is padded to the next power of two.  In doing so we lose the length of the unpadded trace. We should consider persisting the `max_len` as part of the `ExecutionTrace`.

https://github.com/0xPolygonMiden/miden-vm/blob/3022d257eec9cdeca2273b0b66b633bcb3552274/processor/src/trace/mod.rs#L286

**Comment by frisitano** [comment]
Alternatively we could introduce a `TraceInfo` struct that holds this information and make that a field in the `ExecutionTrace` struct.

```rust
pub struct TraceInfo {
     main_trace_len: usize,
     range_table_len: usize,
     hasher_chiplet_len: usize,
     ...
     padded_len: usize
}
```

**Comment by bobbinth** [comment]
Yes, this would be great to add. I like the approach with a more detailed `TraceInfo` struct.

**Comment by bobbinth** [comment]
Closed by #1029.
